src/qililab/qprogram/crosstalk_matrix.py

```diff
@@ -0,0 +1,96 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Collection
+
+from qililab.yaml import yaml
+
+
+@yaml.register_class
+class CrosstalkMatrix:
+    """A class to represent a crosstalk matrix where each index corresponds to a bus."""
+
+    def __init__(self):
+        """Initializes an empty crosstalk matrix."""
+        self.matrix: dict[str, dict[str, float]] = {}
+
+    def __getitem__(self, bus: str) -> dict[str, float]:
+        """Returns the dictionary of crosstalk values for the given bus.
+
+        Args:
+            bus (str): The bus for which to get the crosstalk values.
+
+        Returns:
+            dict[str, float]: A dictionary of crosstalk values for the given bus.
+        """
+        if bus not in self.matrix:
+            self.matrix[bus] = {}
+        return self.matrix[bus]
+
+    def __setitem__(self, key: str, value: dict[str, float]) -> None:
+        """Sets the crosstalk values for the given bus.
+
+        Args:
+            key (str): The bus for which to set the crosstalk values.
+            value (dict[str, float]): A dictionary of crosstalk values to set for the given bus.
+        """
+        self.matrix[key] = value
+
+    def __repr__(self) -> str:
+        """Returns a string representation of the CrosstalkMatrix.
+
+        Returns:
+            str: A string representation of the CrosstalkMatrix.
+        """
+        return f"CrosstalkMatrix({self.matrix})"
+
+    def __str__(self) -> str:
+        """Produces a string representation of the matrix with rows and columns, labels included.
+        The diagonal contains "\", since there is no value there, and if a value doesn't exist, the default 1.0 is shown.
+
+        Returns:
+            str: A string representation of the crosstalk matrix.
+        """
+        buses: set[str] = set(self.matrix.keys())
+        for values in self.matrix.values():
+            buses.update(values.keys())
+
+        sorted_buses = sorted(buses)
+        col_width = max(len(bus) for bus in sorted_buses) + 4  # Determine column width
+        header = " " * col_width + " ".join(f"{bus:>{col_width}}" for bus in sorted_buses) + "\n"
+        rows = []
+        for bus1 in sorted_buses:
+            row = [f"{bus1:{col_width}}"]
+            for bus2 in sorted_buses:
+                if bus1 == bus2:
+                    row.append("\\".rjust(col_width))
+                else:
+                    row.append(f"{self.matrix.get(bus1, {}).get(bus2, 1.0):{col_width}.1f}")
+            rows.append(" ".join(row))
+        return header + "\n".join(rows)
+
+    @classmethod
+    def from_buses(cls, buses: Collection[str]) -> "CrosstalkMatrix":
```

Comment:
The only thing I am missing in this class is a method that returns a matrix in numpy way. But I am not 100% sure that is how @visagim  will prefer it, to perform the linear algebra matrix multiplication, or retrieving it indexed by buses works better for him. I do not mind merging it like this and plan a new task to include that new functionality if Victor tell us he needs it.

Reply:
Yes, I didn't implement this method, since I am not aware of the way that buses will be mapped to the matrix indexes (e.g. `drive_q0_bus` gets assigned to index 0 etc.) I would prefer to merge it as it is, and discuss this mapping in the next sprint with Victor.

Reply:
Sounds good to me.